hypervisor.go

```diff
@@ -190,6 +190,10 @@ type HypervisorConfig struct {
 	// For example, if there is a value for the "kernel" key in this map,
 	// it will be used for the pod's kernel path instead of KernelPath.
 	customAssets map[assetType]*asset
+
+	//Determines how the VM should be connected to the
+	//the container network interface
+	InterNetworkModel NetInterworkingModel
```

Comment:
Why do you need this ?

Reply:
I was looking for a place to be semantically correct for the information provided vc user. 

Actually I would like to ask for feedback for the best place to put it:

Here the context:
From oci pkg we create the `podConfig` we do it based on `RuntimeConfig`
```go
type RuntimeConfig struct {
	VMConfig vc.Resources

	HypervisorType   vc.HypervisorType
	HypervisorConfig vc.HypervisorConfig

	AgentType   vc.AgentType
	AgentConfig interface{}

	ProxyType   vc.ProxyType
	ProxyConfig vc.ProxyConfig

	ShimType   vc.ShimType
	ShimConfig interface{}

	Console string
}
```
https://github.com/containers/virtcontainers/blob/master/pkg/oci/utils.go#L88

From oci  vc user looks good me that the property be placed in:
`config.HypervisorConfig.InterNetworkModel`
Because the property is related on how the Hypervisor will be connected.

On the other hand when the `PodConfig` is created, the struct that mainly carry the network configuration is:
`podConfig.NetworkConfig` so I added the in the property to `podConfig.NetworkConfig.InterNetworkModel`
`networkConfig, err := networkConfig(ocispec, config)`
https://github.com/containers/virtcontainers/pull/528/files#diff-5f81144f5833a1f9b3e10ca1e3039207R258


What I did was to add the property `HypervisorConfig` and then when the `PodConfig` is created use it to properly create/set `podConfig.NetworkConfig.InterNetworkModel`.

But now `config.HypervisorConfig.InterNetworkModel` it is just used to create the network config. `networkConfig, err := networkConfig(ocispec, config)` then is not used anymore.

So I wonder if should I not add it here or added `type RuntimeConfig struct`.

Reply:
@jcvenegas Thanks for this summary, I understand the problem here and I think that we should make a decision in one direction or another, but keeping both ways is too confusing IMO.
I think we should not use `HypervisorConfig` because the VM itself does not need to know about the way the connection is done with the interface inside the network namespace. Instead, I would prefer that we define a new field in the `configuration.toml`, under the runtime section, something like that:
```toml
[runtime]
enable_debug = true
net_xconnect_model = "bridge"
```
From this information, you should be able to provision `podConfig.NetworkConfig` with the correct information.

This is the cleanest solution IMO. @sameo @jodh-intel WDYT ?

Reply:
That sounds good to me, but we might also want to also get agreement from @mcastelino first.

Reply:
Oh yes of course, cc @mcastelino

Reply:
@jcvenegas thanks for doing this. As such it looks good. I wanted to clarify why I added this configurable option. It was added so that we can switch back to the legacy bridge method if we found issues with macvtap. 

Once we found macvtap to be reliable I planned to just remove the bridge networking model. And now we think macvtap has proved to be good enough (except the bug that @jcvenegas for containerd, an issue that should be root caused). So we cannot remove it quite yet :(

So longer term I think we will have this value be configurable for only one reason. 
1. Use tc based interconnection model (as done in runV)
2. Use macvtap

However there are special network interfaces (VFs or Physical NICs), vhost-user-socket) where neither tc or macvtap will work well, and we use an interconnection model based on the type of the incoming interface and not driven by the defined interconnection type. (VFIO, vhost-user).

@sboeuf @egernst added support to get and store the incoming endpoint type. @sboeuf @egernst can you confirm the same please.

So the interconnection model will be used in the case of interfaces where either tc or macvtap can be used. For the end points where neither works, we will connect it to the VM in a way that is specific to the end point.

But if our performance measurements indicate the tc and macvtap give the exact same performance and there is no feature gap in the tc method, I wonder if we should just switch to tc, at which point we remove this. @sameo ?

So the final implementation will be

- tc + multiqueue tap (for normal interfaces)
- end point specific connection for "special endpoints"

/cc @sameo @sboeuf @egernst @jcvenegas @amshinde

Reply:
@mcastelino -- correct -- we do not take the interconnection model into account in the event we are using a VFIO or vhost-user based network interface.  Its less about "work well" and more of "work at all," for these types.

Reply:
@mcastelino I agree with the final implementation, but for the time being, and to be able to switch easily (for testing purpose for instance), I think we can make this interconnect model configurable.
With this in mind, I was saying that I would prefer a new field in our `configuration.toml`, rather than adding this to the `HypervisorConfig`.

Reply:
@sboeuf agreed.

Reply:
@mcastelino  thanks for the clarification. Makes totally sense to me to work to just let one implementation.  For the moment to allow users change it (in any case need, debug, testing, issues with default ). What should be the place to add the option at runtime level or hypervisor level. See comment above from @sboeuf .

Reply:
@jcvenegas I think we all agreed on the same thing here. You can go ahead and modify runtime/virtcontainers PRs so that you will rely on a new field from the configuration.toml to determine which xconnect model you need. We will need a default value in case this model is not provided through the config file.